src/common/ortho.jl

```diff
@@ -1,9 +1,5 @@
-@timing function ortho_qr(φk::ArrayType) where {ArrayType <: AbstractArray}
-    x = convert(ArrayType, qr(φk).Q)
-    if size(x) == size(φk)
-        return x
-    else
-        # Sometimes QR (but funnily not always) CUDA messes up the size here
```

Comment:
@mfherbst do you remember what was going on there? The API for qr is really annoying, because some people use the "full" QR decomposition (with Q being a NxN matrix if A was Nxn), so that this code is possibly a massive hidden performance trap which we should report to CUDA

Reply:
If I recall correctly the sizes it reported (i.e. `size(qr(X)`) were fine, just when you did the `Matrix(qr(X).Q)` then you got a full NxN array back.